### PR TITLE
Ensure that collapsed elements immediately update their layout box

### LINKED ISSUE #4454: Amp-analytics Visibility Spec Doesn't Properly Track Amp-Lightboxes
The amp-analytics component supports the ability to set up triggers that fire when an amp-component has been visible on screen for a defined amount of time. However, this feature is buggy when the trigger is set up on an amp-lightbox. 

When a visibility trigger is set up with the "selector" property pointing to an amp-lightbox, the trigger will correctly deduce when the lightbox becomes visible (when the user has caused the lightbox to open), but amp-analytics will not correctly deduce when the lightbox is closed.

For example, let's say you have a trigger set up on a lightbox that looks something like this: 

`"fiveSecondsHalfVisible": {
          "on": "visible",
          "request": "visibility",
          "visibilitySpec":  {
            "selector": "#lightbox",
            "visiblePercentageMin": 50,
            "totalTimeMin": 5000
      }
}`

Let's say after page load, there is no interaction for 10 seconds. Amp-analytics would correctly not fire the fiveSecondsHalfVisible trigger since the lightbox is closed. Then the user clicks a button that causes the lightbox to open. The user immediately closes the lightbox after opening it by clicking another button. At this time, the lightbox is again not visible. However, after 5 seconds, the fiveSecondsHalfVisible trigger will fire anyway.

You can play with an example of this bug at this [JSBin](http://jsbin.com/fepacukawa/19/edit?html,console,output).

Steps to reproduce:

1. Navigate to [JSBin page](http://jsbin.com/fepacukawa/19/edit?html,console,output).
2. Right click in the "output" pane and open Chrome Dev Tools. 
3. Go to the network tab.
4. Reload the page, keeping the Chrome Dev Tools open.
5. Click on the "Show Lightbox" button.
6. Immediately press the "Hide Lightbox" button.
7. Wait 10 seconds.
8. Observe that there are two requests to "is_visible.com", one for a 5 second event and one for a 10 second event.

The bug is found in AMP Version 1470677001097 and on Chrome.

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #4454

### CODE REVIEW

File: src/service/resource.js [hunk 1]
```diff
@@ -334,6 +335,18 @@ export class Resource {
   }
 
   /**
+   * Completes collapse: ensures that the element is `display:none` and
+   * updates layout box.
+   */
+  completeCollapse() {
+    toggle(this.element, false);
+    this.layoutBox_.width = 0;
```
Comment (jridgewell) [inline src/service/resource.js]: Isn't the layout box immutable? How about setting a new `{ width: 0, height: 0 }`?

Comment (dvoytenko) [inline src/service/resource.js]: Good point. Done.

**Comment by dvoytenko** [comment]
@jridgewell PTAL.

**Comment by jridgewell** [comment]
LGTM.

**Comment by avimehta** [comment]
lgtm

**Pull request merged**